ConfigSpace/hyperparameters.pyx

```diff
@@ -610,6 +611,8 @@ cdef class NormalFloatHyperparameter(FloatHyperparameter):
     def __init__(self, name: str, mu: Union[int, float], sigma: Union[int, float],
                  default_value: Union[None, float] = None,
                  q: Union[int, float, None] = None, log: bool = False,
+                 lower: Union[int, None] = None,
+                 upper: Union[int, None] = None,
```

Comment:
I think it'd be good to go with the explicit `Optional[int]` type here. By definition `Union[int, None] == Optional[int]` but Optional is a bit more syntactically clear